### PR TITLE
ci: run revive using golang-lint-ci

### PR DESCRIPTION
**Pull request by phillebaba**
## Description

This replaces revive with golang-ci-lint, and instead runs revive through the aforementioned tool. Additionally it replaces the header check script with the goheader linter. The benefit of this change is that we can enable more linters along with revive in the future.

## Related Issue

Part of #2503 

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/.github/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: .golangci.yaml
Comment (phillebaba) [inline .golangci.yaml:20]: I disabled linting the Helm tools package as it is a copy of Helm code.

File: .golangci.yaml
Comment (AustinAbro321) [inline .golangci.yaml:10]: I believe revive doesn't lint anything by default. I believe you're going to have to copy the rules from the revive.toml into a linter-settings format unfortunately https://golangci-lint.run/usage/linters/#revive

**Review by AustinAbro321 — CHANGES_REQUESTED** [review]

File: .golangci.yaml
Comment (phillebaba) [reply .golangci.yaml:10]: I checked the revive docs and assumed the opposite.

> Please notice that if no particular configuration is provided, revive will behave as go-lint does, i.e. all go-lint rules are enabled (the [Available Rules table](https://github.com/mgechev/revive?tab=readme-ov-file#available-rules) details what are the go-lint rules). When a configuration is provided, only rules in the configuration are enabled.

https://github.com/mgechev/revive?tab=readme-ov-file#golangci-lint

Comment (phillebaba) [reply .golangci.yaml:10]: Fixed this by enabling rules that are by default ignored by golangci-lint.

https://golangci-lint.run/usage/false-positives/#exc0012

File: .pre-commit-config.yaml
Comment (AustinAbro321) [inline .pre-commit-config.yaml:43]: Looks like when we run this on pre-commit it only takes into account the single file as it will run something like `golangci-lint run src/pkg/packager/deploy.go` which can result in false positives like this.
```
src/pkg/packager/deploy.go:35:10: undefined: Packager (typecheck)
func (p *Packager) resetRegistryHPA() {
         ^
src/pkg/packager/deploy.go:44:10: undefined: Packager (typecheck)
func (p *Packager) Deploy() (err error) {
         ^
src/pkg/packager/deploy.go:123:10: undefined: Packager (typecheck)
func (p *Packager) deployComponents() (deployedComponents []types.DeployedComponent, err error) {
```

We should turn off pass_filesiles and use the --new flag imo

**Review by AustinAbro321 — CHANGES_REQUESTED** [review]

File: .github/workflows/scan-lint.yml
Comment (Noxsios) [inline .github/workflows/scan-lint.yml:14]:
```suggestion
        uses: golangci/golangci-lint-action@a4f60bb28d35aeee14e6880718e0c85ff1882e64 # v6.0.1
```

Comment (Noxsios) [reply .github/workflows/scan-lint.yml:14]: pins so that dependabot / revive updates on sha change

**Review by Noxsios — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml
Comment (phillebaba) [reply .pre-commit-config.yaml]: Took inspiration from another repo and fixed the issue.

https://github.com/golangci/golangci-lint/blob/master/.pre-commit-hooks.yaml

**Comment by phillebaba** [comment]
Just FYI I realized we could remove the requirement to install goimports by just relying on golangci-lint to fix any errors for that specific linter.

File: .pre-commit-config.yaml
Comment (AustinAbro321) [inline .pre-commit-config.yaml:38]: Ready to approve this PR, but thoughts on using the `--new` flag here to speed up lint time? I know it they say here that linters like unused won't work with `--new` but I wasn't able to reproduce that https://github.com/golangci/golangci-lint/blob/master/.pre-commit-hooks.yaml#L3

Comment (phillebaba) [reply .pre-commit-config.yaml:38]: I would rather have all linter rules run, as that is the way it is going to run in the CI. You are not going to save any time if unused would have failed if run, and you will have to wait for the CI to run before you figure that out.

Comment (Noxsios) [reply .pre-commit-config.yaml:39]: @AustinAbro321 for your own local dev, you can always edit your `.pre-commit-config.yaml` to add the `--new` and just not commit it.  Not the best solution, but it should satisfy all parties for now, thoughts?

Comment (AustinAbro321) [reply .pre-commit-config.yaml:39]: Yeah I'm fine with it running without the lint flag just wanted to get Phillips thoughts. Approving now.

**Review by AustinAbro321 — APPROVED** [review]

**Review by Noxsios — APPROVED** [review]

**Pull request merged**
